samples/nrf9160/modem_shell/src/fota/fota.c

```diff
@@ -41,7 +43,7 @@ static void fota_download_callback(const struct fota_download_evt *evt)
 		break;
 	case FOTA_DOWNLOAD_EVT_FINISHED:
 		mosh_print("FOTA: Download finished, rebooting in 5 seconds...");
-		k_timer_start(&reboot_timer, K_SECONDS(5), K_NO_WAIT);
+		k_work_schedule(&system_reboot, K_SECONDS(5));
```

Comment:
Also mosh_common_work_q could be used instead of system workq directly.

Reply:
That's true. Anyway, this only reboots the system and needs a bit less code, so I think this is fine.